src/compiler/Stage/InferTypes/GenerateEquations.elm

```diff
@@ -249,6 +249,12 @@ generateEquations idSource ( expr, type_ ) =
             , idSource2
             )
 
+        Typed.Tuple _ _ ->
+            ( [], idSource )
+
+        Typed.Tuple3 _ _ _ ->
+            ( [], idSource )
```

Comment:
Those need to return the equations for their inner exprs (`generateEquations idSourceX eX`)